designsafe/apps/api/datafiles/operations/agave_operations.py

```diff
@@ -0,0 +1,587 @@
+    """Create a new directory.
+
+    Params
+    ------
+    client: agavepy.agave.Agave
+        Tapis client to use for the listing.
+    system: str
+        Tapis system ID.
+    path: str
+        Path in which to run mkdir.
+    dir_name: str
+        Name of the directory
+
+    Returns
+    -------
+    dict
+    """
+    body = {
+        'action': 'mkdir',
+        'path': dir_name
+    }
+    result = client.files.manage(systemId=system,
+                                 filePath=urllib.parse.quote(path),
+                                 body=body)
+
+    agave_indexer.apply_async(kwargs={'systemId': system,
+                                      'filePath': path,
+                                      'recurse': False},
+                              queue='indexing')
+    return dict(result)
+
+
+def move(client, src_system, src_path, dest_system, dest_path, file_name=None):
+    """Move a current file to the given destination.
+
+    Params
+    ------
+    client: agavepy.agave.Agave
+        Tapis client to use for the listing.
+    src_system: str
+        System ID for the file's source.
+    src_path: str
+        Path to the source file.
+    dest_system: str
+        System ID for the destination.
+    dest_path: str
+        Path under which the file should be moved.
+    file_name: str
+        New name for the file if desired.
+
+    Returns
+    -------
+    dict
+
+    """
+
+    if file_name is None:
+        file_name = src_path.strip('/').split('/')[-1]
+
+    dest_path_full = os.path.join(dest_path.strip('/'), file_name)
+    src_path_full = urllib.parse.quote(src_path)
+
+    # Handle attempt to move a file into its current path.
+    if src_system == dest_system and src_path_full == dest_path_full:
+        return {'system': src_system, 'path': src_path_full, 'name': file_name}
+
+    try:
+        client.files.list(systemId=dest_system,
+                          filePath="{}/{}".format(dest_path, file_name))
+
+        # Destination path exists, must make it unique.
+        _ext = os.path.splitext(file_name)[1].lower()
+        _name = os.path.splitext(file_name)[0]
+        now = datetime.datetime.utcnow().strftime('%Y-%m-%d %H-%M-%S')
+        file_name = '{}_{}{}'.format(_name, now, _ext)
+    except HTTPError as err:
+        if err.response.status_code != 404:
+            raise
+
+    full_dest_path = os.path.join(dest_path.strip('/'), file_name)
+
+    if src_system == dest_system:
+        body = {'action': 'move',
+                'path': full_dest_path}
+        move_result = client.files.manage(systemId=src_system,
+                                          filePath=urllib.parse.quote(
+                                              src_path),
+                                          body=body)
+    else:
+        src_url = 'agave://{}/{}'.format(
+            src_system,
+            urllib.parse.quote(src_path)
+        )
+        move_result = client.files.importData(
```

Comment:
this is the async importData followed by the files delete that caused us problems elsewhere.  do we need avoid that here with an exception or have use the agavepy-based solution of waiting for the operation to finish?

Reply:
Good catch. I need to add that agavepy polling operator back in since users might expect inter-system move to still work.